internal/k8s/statusaddress_test.go

```diff
@@ -341,6 +345,254 @@ func TestStatusAddressUpdater(t *testing.T) {
+					},
+				},
+			},
+		},
+		"Gateway not ready": {
+			status:                     ipLBStatus,
+			gatewayClassControllerName: "projectcontour.io/contour",
+			preop: &gatewayapi_v1alpha2.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "projectcontour",
+					Name:      "contour-gateway",
+				},
+				Spec: gatewayapi_v1alpha2.GatewaySpec{
+					GatewayClassName: gatewayapi_v1alpha2.ObjectName("contour-gatewayclass"),
+				},
+				Status: gatewayapi_v1alpha2.GatewayStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   string(gatewayapi_v1alpha2.GatewayConditionReady),
+							Status: metav1.ConditionFalse,
+						},
+					},
+				},
+			},
+			postop: &gatewayapi_v1alpha2.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "projectcontour",
+					Name:      "contour-gateway",
+				},
+				Spec: gatewayapi_v1alpha2.GatewaySpec{
+					GatewayClassName: gatewayapi_v1alpha2.ObjectName("contour-gatewayclass"),
+				},
+				Status: gatewayapi_v1alpha2.GatewayStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   string(gatewayapi_v1alpha2.GatewayConditionReady),
+							Status: metav1.ConditionFalse,
+						},
+					},
+				},
+			},
+		},
+		"Gateway not controlled by this Contour": {
+			status:                     ipLBStatus,
+			gatewayClassControllerName: "projectcontour.io/some-other-controller",
+			preop: &gatewayapi_v1alpha2.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "projectcontour",
+					Name:      "contour-gateway",
+				},
+				Spec: gatewayapi_v1alpha2.GatewaySpec{
+					GatewayClassName: gatewayapi_v1alpha2.ObjectName("contour-gatewayclass"),
+				},
+				Status: gatewayapi_v1alpha2.GatewayStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   string(gatewayapi_v1alpha2.GatewayConditionReady),
+							Status: metav1.ConditionTrue,
+						},
+					},
+				},
+			},
+			postop: &gatewayapi_v1alpha2.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "projectcontour",
+					Name:      "contour-gateway",
+				},
+				Spec: gatewayapi_v1alpha2.GatewaySpec{
+					GatewayClassName: gatewayapi_v1alpha2.ObjectName("contour-gatewayclass"),
+				},
+				Status: gatewayapi_v1alpha2.GatewayStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   string(gatewayapi_v1alpha2.GatewayConditionReady),
+							Status: metav1.ConditionTrue,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name+" OnAdd", func(t *testing.T) {
+			suc := StatusUpdateCacher{}
+			assert.True(t, suc.Add(tc.preop.Name, tc.preop.Namespace, tc.preop), "unable to add object to cache")
+
+			mockCache := &mocks.Cache{}
+			mockCache.
+				On("Get", mock.Anything, client.ObjectKey{Name: string(tc.preop.Spec.GatewayClassName)}, mock.Anything).
+				Run(func(args mock.Arguments) {
+					args[2].(*gatewayapi_v1alpha2.GatewayClass).Spec.ControllerName = gatewayapi_v1alpha2.GatewayController(tc.gatewayClassControllerName)
```

Comment:
nit: maybe add a comment for what this is doing? since `Get` mutates the passed in pointer (similar to below for `OnUpdate`)

Reply:
:+1: added